tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/ClassWriter.java

```diff
@@ -210,7 +210,7 @@ private static String writeGeneratedAnnotation(MetaEntity entity, Context contex
 	}
 
 	private static String writeSuppressWarnings() {
-		return "@SuppressWarnings(\"all\")";
+		return "@SuppressWarnings({ \"deprecation\", \"rawtypes\", \"serial\" })";
```

Comment:
Thanks for the tip. I have one question. Why did you pick these [3 our of all supported warnings](https://docs.oracle.com/javase/7/docs/technotes/tools/solaris/javac.html#xlintwarnings)?

I wonder if we should also add  `varargs` to the list.

Reply:
Admittedly I'm not exactly sure which ones can occur. We know we need `"deprecation"` and `"rawtypes"`, but I'm not sure if/when others may occur.

I don't think it's possible for a metamodel class to implement `Serializable` so I don't think we need `"serial"`. I'll update the pull request.

And if the only thing possible in a generated metamodel class is the `public static volatile ...` declarations (appears to be the case from looking at `ClassWriter.generateBody()`), then I don't think any other warnings could occur.

Re `"varargs"` .. how could that happen? I don't see it but could be missing something.